app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,797 @@
+            }
+          },
+          {
+            name: 'Colon',
+            onClick: function () {
+              Actions.setParseConfig('delimiter', ':');
+            }
+          },
+          {
+            name: 'Hyphen',
+            onClick: function () {
+              Actions.setParseConfig('delimiter', '-');
+            }
+          },
+        ]
+      };
+      return <Components.SmallDropdown dropdownSetup={setup}/>;
+    },
+
+    renderControls: function () {
+      if (this.props.filesize < 200000) {
+        return (
+          <span>
+            {this.header()}
+            {this.numbersFormat()}
+            {this.delimiter()}
+          </span>
+        );
+      }
+
+      return (
+        <p className="no-options-available">
+          Fine grained import options are disabled for files exceeding 200KB
+        </p>
+      );
+    },
+
+    render: function () {
+      var controls = this.renderControls();
+
+      return (
+        <div className="import-options-row">
+          <div className="options-row">
+            {this.previewToggle()}
+            {controls}
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var TableView = React.createClass({
+    eachRow: function () {
+      var data = this.props.data;
+
+      if (this.props.isBigFile) {
+        data = this.props.getSmallPreviewOfData;
+      }
+
+      return (
+        data.map(function (dataObj, i) {
+          if (i < 500) {
+            return <tr key={i}>{this.insideEachRow(dataObj)}</tr>;
+          }
+        }.bind(this))
+      );
+    },
+
+    insideEachRow: function (dataObj) {
+      return _.map(dataObj, function (dataVal, dataKey) {
+        return <td key={dataKey} title={dataVal}>{dataVal}</td>;
+      });
+    },
+
+    header: function () {
+      var header = null;
+
+      if (this.props.getHeaderConfig) {
+        header = this.props.meta.fields;
+        return (
+          header.map(function (field, i) {
+            return <th key={i} title={field}>{field}</th>;
+          })
+        );
+      } else {
+        header = this.props.data;
+        return (
+          header.map(function (field, i) {
+           return <th key={i} title={i}>{i}</th>;
+          })
+        );
+      }
+    },
+
+    render: function () {
+      var data = this.eachRow(),
+          header = this.header();
+
+      if (this.props.getPreviewView != 'table') {
+        return null;
+      }
+      return (
+        <table className="data-import-table">
+          <tbody>
+            <tr>{header}</tr>
+            {data}
+          </tbody>
+        </table>
+      );
+    }
+  });
+
+  var JSONView = React.createClass({
+    objectify: function (array) {
+      return _.reduce(array, function (obj, val, i) {
+        obj[i] = val;
+        return obj;
+      }, {});
+    },
+
+    rows: function () {
+      var data = this.props.data;
+
+      if (this.props.isBigFile) {
+        data = this.props.getSmallPreviewOfData;
+      }
+
```

Comment:
you can make this decision (which data to use) already in the store. 

use one `getData()` getter and the store then returns the right data. this keeps the views dumb and makes testing easier.